dotnet/src/Connectors/Connectors.Memory.Redis/RedisCollectionSearchMapping.cs

```diff
@@ -78,10 +78,11 @@ internal static Query BuildQuery<TRecord>(Expression<Func<TRecord, bool>> filter
             .Limit(options.Skip, top)
             .Dialect(2);
 
-        var sortInfo = options.OrderBy.Values.Count switch
+        var orderByValues = options.OrderBy?.Invoke(new()).Values;
+        var sortInfo = orderByValues switch
         {
-            0 => null,
-            1 => options.OrderBy.Values[0],
+            null => null,
+            _ when orderByValues.Count == 1 => orderByValues[0],
```

Comment:
nit: list pattern matching FTW (also in the other connectors):

```suggestion
            [var singleOrderBy] => singleOrderBy,
```

Reply:
It looks better, but... anytime you run the full build the dotnet format tool is going to update the file and mess the indentation LOL

At least this is something that always happen to me locally when I run `build.cmd`

![{D122F4BA-D2DA-45E0-B8FF-296481C8E8CC}](https://github.com/user-attachments/assets/241cb979-346f-40cb-b1ab-02686dfe582a)

Reply:
Really? Weird... Looks like a bug we should maybe file...